### ISSUE TITLE
Fatal error in node startup while using state-sync-enabled = true in 1.7.11

### ISSUE THREAD

**Issue by shubhamgoyal42** [ava-labs/avalanchego#1074]
**Describe the bug**
Node exists after FATAL error

**To Reproduce**
1. Use avalanche/1.7.11
2. Create the file `~/.avalanchego/configs/chains/C/config.json` and add the following lines
```
{
        "state-sync-enabled": true
}
```
3. Start the node on mainnet, and let it bootstrap. It finished for me after 36-40 hours.
4. Restart the node. At this point the fatal error will occur. Example log:
```
[06-03|06:39:23.858] FATAL <C Chain> handler/handler.go:274 shutting down chain due to unexpected error: failed to notify VM that bootstrapping has started: proposervm height index (15532032) should never be lower than the inner height index (15535131) while processing sync message: (Op: get_accepted_state_summary_failed, NodeID: NodeID-NaDQPxLhuMhmYvaw7zxq6nRoncPzFcwLE, RequestID: 2)
```

**Expected behavior**
Node to start normally and run the pending executions.

**Screenshots**
![Screenshot 2022-06-03 at 12 35 43 PM](https://user-images.githubusercontent.com/22051142/171804718-44a06f2c-f2fa-44e7-b57c-5f3180c0e2cb.png)

**Logs**
```
[06-03|06:38:32.723] INFO <C Chain> syncer/state_syncer.go:332 Starting state sync...
[06-03|06:38:32.723] INFO chains/manager.go:263 creating chain:
    ID: 2oYMBNV4eNHyqk2fjjV5nVQLDbtmNJzq5s3qs3Lo6ftnC6FByM
    VMID:jvYyfQTxGMJLuGWa55kdP2p2zSUYsQ5Raupu4TW34ZAUBAbtq
[06-03|06:38:32.741] INFO <X Chain> avm/vm.go:453 Fee payments are using Asset with Alias: AVAX, AssetID: FvwEAhmxKfeiG8SnEvq42hc6whRyY3EFYAvebMqDNDGCgxN5Z
[06-03|06:38:32.741] INFO <X Chain> avm/vm.go:235 address transaction indexing is disabled
[06-03|06:38:32.744] INFO <X Chain> avalanche/transitive.go:73 initializing consensus engine
[06-03|06:38:32.745] INFO server/server.go:237 adding route /ext/bc/2oYMBNV4eNHyqk2fjjV5nVQLDbtmNJzq5s3qs3Lo6ftnC6FByM
[06-03|06:38:32.746] INFO server/server.go:237 adding route /ext/bc/2oYMBNV4eNHyqk2fjjV5nVQLDbtmNJzq5s3qs3Lo6ftnC6FByM/wallet
[06-03|06:38:32.746] INFO server/server.go:237 adding route /ext/bc/2oYMBNV4eNHyqk2fjjV5nVQLDbtmNJzq5s3qs3Lo6ftnC6FByM/events
[06-03|06:38:32.746] INFO <X Chain> bootstrap/bootstrapper.go:264 Starting bootstrap...

[06-03|06:38:38.836] INFO <C Chain> syncer/state_syncer.go:355 starting state sync
[06-03|06:38:52.610] INFO <X Chain> common/bootstrapper.go:224 Bootstrapping started syncing with 1 vertices in the accepted frontier
[06-03|06:38:56.197] INFO <X Chain> bootstrap/bootstrapper.go:477 bootstrapping fetched 1 vertices. Executing transaction state transitions...
[06-03|06:38:56.214] INFO <X Chain> queue/jobs.go:182 executed 1 operations
[06-03|06:38:56.214] INFO <X Chain> bootstrap/bootstrapper.go:488 executing vertex state transitions...
[06-03|06:38:56.214] INFO <X Chain> queue/jobs.go:182 executed 1 operations

[06-03|06:39:23.857] INFO <C Chain> syncer/state_syncer.go:274 Selected summary Wcodi8N2zGUK2ck2Z4ALeTBZ6F9baCAdokLPCxjPqPM7ekxNk out of 1 to start state sync
INFO [06-03|06:39:23.857] <C Chain> github.com/ava-labs/coreth/plugin/evm/syncervm_client.go:179: last accepted too close to most recent syncable block, skipping state sync lastAccepted=15,535,131 syncableHeight=15,532,032
INFO [06-03|06:39:23.858] <C Chain> github.com/ava-labs/coreth/core/blockchain.go:281: Initializing snapshots async=true
[06-03|06:39:23.858] INFO <C Chain> bootstrap/bootstrapper.go:112 Starting bootstrap...
[06-03|06:39:23.858] FATAL <C Chain> handler/handler.go:274 shutting down chain due to unexpected error: failed to notify VM that bootstrapping has started: proposervm height index (15532032) should never be lower than the inner height index (15535131) while processing sync message: (Op: get_accepted_state_summary_failed, NodeID: NodeID-NaDQPxLhuMhmYvaw7zxq6nRoncPzFcwLE, RequestID: 2)
[06-03|06:39:23.858] INFO <C Chain> bootstrap/bootstrapper.go:256 shutting down bootstrapper
INFO [06-03|06:39:23.858] <C Chain> github.com/ava-labs/coreth/core/tx_pool.go:457: Transaction pool stopped
INFO [06-03|06:39:23.858] <C Chain> github.com/ava-labs/coreth/core/blockchain.go:522: Shutting down state manager
INFO [06-03|06:39:23.859] <C Chain> github.com/ava-labs/coreth/core/blockchain.go:528: Shutting down sender cacher
INFO [06-03|06:39:23.859] <C Chain> github.com/ava-labs/coreth/core/blockchain.go:532: Closing scope
INFO [06-03|06:39:23.859] <C Chain> github.com/ava-labs/coreth/core/blockchain.go:535: Blockchain stopped
[06-03|06:39:23.859] INFO node/node.go:1211 shutting down node with exit code 1
[06-03|06:39:23.859] INFO ipcs/chainipc.go:101 shutting down chain IPCs
[06-03|06:39:23.859] INFO chains/manager.go:974 shutting down chain manager
[06-03|06:39:23.859] INFO router/chain_router.go:252 shutting down chain router
[06-03|06:39:23.859] INFO <P Chain> bootstrap/bootstrapper.go:256 shutting down bootstrapper
[06-03|06:39:23.859] INFO <X Chain> bootstrap/bootstrapper.go:257 shutting down bootstrapper
[06-03|06:39:23.859] INFO network/network.go:1021 shutting down the p2p networking
[06-03|06:39:23.898] INFO node/node.go:1254 cleaning up plugin subprocesses
[06-03|06:39:23.899] INFO node/node.go:1257 finished node shutdown
[06-03|06:39:23.909] INFO nat/nat.go:152 Unmapped all ports```

**Operating System**
Ubuntu 20.04 LTS

**Additional context**
I used the new feature of state-sync as explained here - https://twitter.com/_patrickogrady/status/1529275427708997632

**Comment by shubhamgoyal42** [comment]
Also reported on discord:

https://discord.com/channels/578992315641626624/620633143002660874/982172331554275348

**Comment by StephenButtolph** [comment]
This will be resolved in v1.7.12

**Comment by StephenButtolph** [comment]
Closing this now that v1.7.12 is out. Note, any DB that ran into this issue in v1.7.11 is corrupted. Updating to v1.7.12 is not expected to repair the database.
